src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -97,9 +131,34 @@ public CommandResult addUserTags(Model model) throws CommandException {
         model.getObservablePersonList().forEach(person ->
                 person.setCommonModules(editedUser.getImmutableModuleTags()));
 
+        for (ModuleTag tag : moduleTags) {
+            String day = tag.getDay();
+            String start = tag.getStartTime();
+            String end = tag.getEndTime();
+            if (day == null || start == null || end == null) {
+                continue;
+            }
+
+            Module mod = new Module(tag.toString());
+            int startHour = Integer.parseInt(start.substring(0, 2));
```

Comment:
If you notice that you’re computing the start and end times a lot, why not extract into a separate method?